src/data_source/mod.rs

```diff
@@ -29,6 +29,7 @@ mod hdbc_wrapper;
 pub struct DataSource<'env, S: HDbcWrapper<'env> = Unconnected<'env>> {
     /// Connection handle. Either `HDbc` for `Unconnected` or `Connected` for `Connected`.
     handle: S::Handle,
+    autocommit: bool
```

Comment:
I do not like the idea of having extra members on the handle wrappers. For once it introduces extra invariants to maintain: Extra care must be taken to ensure that the autocommit state matches the handle.

The autocommit state can retrieved via `SQL_ATTR_AUTOCOMMIT ` anyway.

In addition to that the idea of autocomitting does not fit well with the concept of moddeling state transitions within the type system.